clash-lib/prims/verilog/Clash_Sized_Vector.primitives.yaml

```diff
@@ -136,23 +136,22 @@
 - BlackBox:
     name: Clash.Sized.Vector.imap_go
     kind: Declaration
-    type: 'imap :: Index n
-      -> (Index n -> a -> b) -> Vec m a -> Vec m b'
+    type: 'imap_go :: (Index n -> a -> b) -> Vec m a -> Index n -> Vec m b'
     template: |-
       // imap begin
       genvar ~GENSYM[i][1];
       ~GENERATE
       for (~SYM[1]=0; ~SYM[1] < ~LENGTH[~TYPO]; ~SYM[1] = ~SYM[1] + 1) begin : ~GENSYM[imap][2]
-        wire ~TYP[0] ~GENSYM[map_index][3];~IF~SIZE[~TYP[2]]~THEN
-        wire ~TYPEL[~TYP[2]] ~GENSYM[map_in][4];
-        assign ~SYM[4] = ~VAR[vec][2][~SYM[1]*~SIZE[~TYPEL[~TYP[2]]]+:~SIZE[~TYPEL[~TYP[2]]]];~ELSE ~FI
-        ~OUTPUTUSAGE[1] ~TYPEL[~TYPO] ~GENSYM[map_out][5];
+        wire ~TYP[2] ~GENSYM[map_index][3];~IF~SIZE[~TYP[1]]~THEN
+        wire ~TYPEL[~TYP[1]] ~GENSYM[map_in][4];
+        assign ~SYM[4] = ~VAR[vec][2][~SYM[1]*~SIZE[~TYPEL[~TYP[1]]]+:~SIZE[~TYPEL[~TYP[1]]]];~ELSE ~FI
```

Comment:
```suggestion
        assign ~SYM[4] = ~VAR[vec][1][~SYM[1]*~SIZE[~TYPEL[~TYP[1]]]+:~SIZE[~TYPEL[~TYP[1]]]];~ELSE ~FI
```
Right?

> `~VAR[<NAME>][N]`: Like `~ARG[N]` ...

Reply:
aagh yes.. forget the vars. It's indeed worrying that that test catches basically nothing..

Reply:
I also forgot about `~TYP` :(

Reply:
I think you're misreading my _Suggested change_, because it looks to me like you got the `~TYP`. It was 2 in the old version, you corrected it to 1, and then I _added_ a correction on top. But that means your corrected version ends up with a red background.

Reply:
oooh of course